review page Baking (#360)
trilitech/tezos-developer-docs: docs/architecture/baking.md

  date: 21 February 2024
---

Baking is the process of creating new blocks in the Tezos blockchain.
Bakers are daemons complementary to Tezos nodes that cooperate to achieve consensus about the next block to add.

> For doing so, bakers validate pending operations, package them into a block, sign the block, propose the new block to other nodes, and verify that the blocks that other bakers propose are valid.

Reviewer: ```suggestion
Bakers validate pending operations, package them into a block, sign the block, propose the new block to other nodes, and verify that the blocks that other bakers propose are valid.
```

> Baker daemons are run on behalf of user accounts which stake tez to guarantee honest participation and receive rewards for their participation.

Reviewer: The chain of phrases in this sentence is a little hard to follow. How about:
```suggestion
To run a baker, a user account must stake tez to guarantee honest participation and receive baking rewards.
```

Author: see next thread for a suggestion to turn this active voice

> By extension, bakers also denote the users running baker daemons on behalf of their user accounts.

Reviewer: Not sure what this means. Maybe "Each baker daemon is connected to a single user account."?

## The baking process

The baking process includes many steps that Tezos users and developers don't need to think about, but at a basic level, baking a block follows these steps:

> 1. The protocol selects in advance a list of bakers for each block to create in a _cycle_, which is a certain number of blocks.

Reviewer: ```suggestion
1. Before each _cycle_, the protocol selects a list of bakers to generate the blocks in that cycle.
```

1. It also selects a list of bakers to act as validators.
These bakers are responsible for validating the blocks that other bakers create and publishing _attestations_ that the block is valid.
1. The first baker on the list has a certain amount of time to create the next block.
It packages pending operations into a block, signs it, and distributes it to the validators and other nodes.
1. If the first baker doesn't publish a block in time, the next baker on the list has a certain amount of time to bake a block, and so on until a block is created.
1. The validators verify the block and publish their attestations in future blocks.

> 1. The next list of bakers create the next block, until the end of the cycle.

Reviewer: ```suggestion
1. The baker on the list creates the next block, and so on until the end of the cycle.
```

"The next list of bakers" suggests that there is another list. Isn't it just one list of bakers for the cycle?

Author: No.
